### PR TITLE
8297822: De-duplicate code in module jdk.sctp

### PR DESCRIPTION
**Pull request by minborg**
This PR proposes merging logic and optimising three classes that exist for aix, macos and windows.

Optimisation will reduce byte code. Below is an example for one of the many methods optimised.

Before:
```
public void implCloseSelectableChannel() throws java.io.IOException;
  Code:
     0: new           #7                  // class java/lang/UnsupportedOperationException
     3: dup
     4: ldc           #11                 // String SCTP not supported on this platform
     6: invokespecial #13                 // Method java/lang/UnsupportedOperationException."<init>":(Ljava/lang/String;)V
     9: athrow
```

After:
```
public void implCloseSelectableChannel() throws java.io.IOException;
  Code:
     0: invokestatic  #16                 // Method sun/nio/ch/sctp/UnsupportedUtil.sctpUnsupported:()Ljava/lang/UnsupportedOperationException;
     3: athrow
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297822](https://bugs.openjdk.org/browse/JDK-8297822): De-duplicate code in module jdk.sctp

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**) ⚠️ Review applies to [6c5c1f41](https://git.openjdk.org/jdk/pull/11436/files/6c5c1f41d59c823defa66853386fb833aa6afb98)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11436/head:pull/11436` \
`$ git checkout pull/11436`

Update a local copy of the PR: \
`$ git checkout pull/11436` \
`$ git pull https://git.openjdk.org/jdk pull/11436/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11436`

View PR using the GUI difftool: \
`$ git pr show -t 11436`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11436.diff">https://git.openjdk.org/jdk/pull/11436.diff</a>

</details>

### CODE REVIEW

File: src/jdk.sctp/share/classes/sun/nio/ch/sctp/UnsupportedUtil.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package sun.nio.ch.sctp;
```
Comment (dfuch) [inline src/jdk.sctp/share/classes/sun/nio/ch/sctp/UnsupportedUtil.java:1]: Still need regular copyright header :-)

**Review by dfuch — COMMENTED** [review]
Good to see all the duplication go away.

**Comment by AlanBateman** [comment]
As the Solaris port is removed then it means the only implementation is on Linux. So one thing to try to is to rename src/jdk.sctp/unix to src/jdk.sctp/linux, src/jdk.sctp/windows/classes/sun/nio/ch/sctp/*  tosrc/jdk.sctp/share/classes/sun/nio/ch/sctp/*  and remove the src/jdk.sctp/macos and src/jdk.sctp/aix tree. If it works then it would avoid needing to have 3 copies of the "unsupported" implementation and avoid include UnsupportedXXX classes in the Linux implementation.

File: src/jdk.sctp/macosx/classes/sun/nio/ch/sctp/SctpChannelImpl.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2009, 2013, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2009, 2012, Oracle and/or its affiliates. All rights reserved.
```
Comment (RogerRiggs) [inline src/jdk.sctp/macosx/classes/sun/nio/ch/sctp/SctpChannelImpl.java:2]: Please check the copyrights (2022), some look like they went back in time.

**Review by dfuch — APPROVED** [review]

**Comment by dfuch** [comment]
Please make sure that all related tests (as well as tier1 and tier2) pass before integrating.

**Review by RogerRiggs — APPROVED** [review]
Looks much better

**Comment by AlanBateman** [comment]
The latest update looks quite good but you might want to update `make/jdk.sctp/Java.gmk` to exclude `UnsupportedUtils` on linux, meaning there is no need to compile/include that on a Linux build.

**Comment by dfuch** [comment]
Hi Per, I will sponsor this PR after the RDP 1 fork.

**Comment by minborg** [comment]
Can either of you please have a look at the make file @AlanBateman or @RogerRiggs  ?

**Comment by AlanBateman** [comment]
> Can either of you please have a look at the make file @AlanBateman or @RogerRiggs ?

It looks right. SCTP used to be Solaris + Linux, now it's Linux only. If there is a low chance that it will remain the only port then we could move it to src/jdk.sctp/linux but no need to do that now.

**Review by RogerRiggs — APPROVED** [review]
LGTM

**Comment by magicus** [comment]
Please note that Skara does not update the list of labels when new files are added to a PR after it's creation. It is up to the committer and reviewer to be aware of such changes during the lifetime of a PR. In this case, it would have been appropriate to add the `build` label when the changes to the makefiles were introduced.

**Pull request closed without merging**